### ISSUE TITLE
Computing gradient when using tf.split generates console log

### ISSUE THREAD

**Issue by eddiepierce** [tensorflow/tensorflow#9894]
### System information
- **Have I written custom code**: No
- **OS Platform and Distribution**: Arch Linux
- **TensorFlow installed from**: source
- **TensorFlow version**: 1.1.0
- **CUDA/cuDNN version**: 8, 6
- **GPU model and memory**: Titan X, 12 GB

### Describe the problem
When computing gradient and using tf.split with size_splits (not num_splits), tensorflow generates console log

### Source code / logs

```

import tensorflow as tf

SIZE = 1

x = tf.random_normal([1, 1])
w = tf.get_variable("w", [1, SIZE*2])
g = tf.matmul(x, w)
s1, s2 = tf.split(g, [SIZE, SIZE], 1) # CONSOLE MESSAGE
# s1, s2 = tf.split(g, 2, 1) # NO CONSOLE MESSAGE
y = s1 * s2

loss = tf.reduce_mean(tf.square(x - y))
train_op = tf.train.AdamOptimizer().minimize(loss)

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    sess.run(train_op)

```
Console message looks like:
[<tf.Tensor 'gradients/split_grad/concat:0' shape=(1, 2) dtype=float32>, None, None]

**Comment by appierys** [comment]
I have the same problem. Moreover, the gradients computed may be corrupted. Although I am not sure actually.

**Comment by concretevitamin** [comment]
@eddiepierce: can you paste the complete log?  Is this a harmless info log?

**Comment by eddiepierce** [comment]
@concretevitamin 
```
>> python bug.py
[<tf.Tensor 'gradients/split_grad/concat:0' shape=(1, 2) dtype=float32>, None, None]
2017-05-15 12:02:08.328921: I tensorflow/core/common_runtime/gpu/gpu_device.cc:887] Found device 0 with properties: 
name: GeForce GTX TITAN X
major: 5 minor: 2 memoryClockRate (GHz) 1.2405
pciBusID 0000:03:00.0
Total memory: 11.92GiB
Free memory: 360.44MiB
2017-05-15 12:02:08.328942: I tensorflow/core/common_runtime/gpu/gpu_device.cc:908] DMA: 0 
2017-05-15 12:02:08.328947: I tensorflow/core/common_runtime/gpu/gpu_device.cc:918] 0:   Y 
2017-05-15 12:02:08.328951: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:03:00.0)
```

"Free memory" is low because I'm running a separate model. 

Yes, it's a harmless info log.

**Comment by concretevitamin** [comment]
@ekelsen: at a casual glance I didn't find any print statement or logging in `split`/`split_v`-related code (and their gradients).  Do you know where that gradient array is printed?

**Comment by skye** [comment]
I tried this myself, the print statement occurs when you run `train_op = tf.train.AdamOptimizer().minimize(loss)` (specifically the minimize() call). I'm digging into finding where the errant print is.

**Comment by skye** [comment]
It's here: https://github.com/tensorflow/tensorflow/blob/r1.1/tensorflow/python/ops/array_grad.py#L274
It looks like this is already fixed in master: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/array_grad.py#L271

Thanks for reporting!

**Comment by LucasMahieu** [comment]
Sorry, I saw that this is closed but I have the save console log message.
I really don't understand the reason of this message ...
Do you think the retro propagation can be corrupted by this ?

**Comment by skye** [comment]
@LucasMahieu what version of TensorFlow are you running?

**Comment by LucasMahieu** [comment]
1.1

-------------------------
Lucas MAHIEU
Lucas.mahieu@icloud.com

**Comment by skye** [comment]
If you upgrade to 1.2 the message should go away. It has no other effect besides printing to the console though.

**Comment by johnarseal** [comment]
Hi,

Now that even I have upgraded to 1.8, the problem is still with me.  

`
conv4_1/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv3_3/biases/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv3_3/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv3_2/biases/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv3_2/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv3_1/biases/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv3_1/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv2_2/biases/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv2_2/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv2_1/biases/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv2_1/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv1_2/biases/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv1_2/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv1_1/biases/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
conv1_1/weights/initial_value: (Const): /job:localhost/replica:0/task:0/device:GPU:0
`  

Part of the console log looks like above. The log gets print whenever I called optimizer.compute_gradients()  
I know it is harmless though, but the log is so long (longer when the model is deeper, because for every variable it prints a line) that it overwhelms all my own self-written debug printing message on the screen. It has been really a big trouble. Can anyone fix it?
